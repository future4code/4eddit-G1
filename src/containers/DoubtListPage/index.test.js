import React from "react";
import { shallow } from "enzyme";
import { DoubtList, SendIconStyled} from "./";
import {UpArrow, DownArrow} from "./styled";
import renderer from "react-test-renderer";
  
describe("DoubtList item", () => {
    test("DoubtList create post", () => {
        const mockGenericFunction = jest.fn()
        const component = shallow(

            
        <DoubtList  allPosts= {[]} getPosts={mockGenericFunction} setErrorMsg={mockGenericFunction} createPost={mockGenericFunction} goToHomePage={mockGenericFunction}/>
          );

          const instance = component.instance();

          instance.onClickCreatePost= jest.fn()

        const sendIconStyled = component.find(SendIconStyled);

        expect(sendIconStyled).toHaveLength(1);

        sendIconStyled.simulate("click");


    })

})

describe("SendButton", () => {
  // Mesma coisa do que test()
  it("should match snapshot", () => {
    const tree = renderer.create(<SendIconStyled />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});