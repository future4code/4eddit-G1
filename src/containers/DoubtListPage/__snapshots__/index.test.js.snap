// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SendButton should match snapshot 1`] = `
<svg
  aria-hidden="true"
  className="MuiSvgIcon-root-1 sc-ksYbfQ cpDAlq"
  focusable="false"
  role="presentation"
  viewBox="0 0 24 24"
>
  <path
    d="M2.01 21L23 12 2.01 3 2 10l15 2-15 2z"
  />
</svg>
`;
